zebrad/tests/common/lightwalletd/send_transaction_test.rs

```diff
@@ -195,20 +265,29 @@ pub async fn run() -> Result<()> {
 
         tracing::info!(?transaction_hash, "sending transaction...");
 
-        let request = prepare_send_transaction_request(transaction);
-
-        let response = rpc_client.send_transaction(request).await?.into_inner();
+        let request = prepare_send_transaction_request(transaction.clone());
+
+        match rpc_client.send_transaction(request).await {
+            Ok(response) => assert_eq!(response.into_inner(), expected_response),
+            Err(err) => {
+                tracing::warn!(?err, "failed to send transaction");
+                zebrad_rpc_client
+                    .send_transaction(transaction)
+                    .await
+                    .map_err(|e| eyre!(e))?;
+            }
```

Comment:
What is the reasoning for this? Wouldn't it be better to fail the test if sending a tx through the rpc_client fails?

Reply:
This was meant to help with debugging since lightwalletd was failing to parse the error.